server/middlewares/async.ts

```diff
@@ -31,9 +32,31 @@ function asyncRetryTransactionMiddleware (fun: (req: Request, res: Response, nex
   }
 }
 
+function executeIfMethod (fun: (req: Request, res: Response, next: NextFunction) => void | Promise<void>, method: string) {
+  return (req: Request, res: Response, next: NextFunction) => {
+    return req.method === method
+      ? Promise.resolve((fun as RequestHandler)(req, res, next))
+      : next()
+  }
+}
+
+function executeIfPOST (fun) {
+  return executeIfMethod(fun, 'POST')
+}
+
+function onlyAllowMethods (methods: string[]) {
```

Comment:
I wasn't able to solve this, I got type errors in the method since `req.method` is typed as `string`. I'm novice to typescript so maybe that's possible to solve.

Reply:
I'll handle it @kontrollanten 😉